src/EmbedIO/Sessions/SessionProxy.cs

```diff
@@ -83,12 +83,7 @@ public DateTime LastActivity
         /// <inheritdoc/>
         public void Delete()
         {
-            EnsureSessionExists();
-
-            if (_session == null)
-                return;
-
-            _sessionManager!.Delete(_context, _session.Id);
+            _sessionManager!.Delete(_context);
```

Comment:
Are you implying that the bug was here, because of the non-atomicity of this operation?
Maybe you are right (and the code certainly looks better with your modifications) but I don't think it would explain the behavior completely. There is certainly no race condition in the failing unit test.